src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -299,6 +316,66 @@ def enable_zip_deploy(cmd, resource_group_name, name, src, timeout=None, slot=No
     return response
 
 
+def upload_zip_to_storage(cmd, resource_group_name, name, src, slot=None):
+    settings = get_app_settings(cmd, resource_group_name, name, slot)
+
+    storage_connection = None
+    for keyval in settings:
+        if keyval['name'] == 'AzureWebJobsStorage':
+            storage_connection = str(keyval['value'])
+
+    if storage_connection is None:
+        raise CLIError('Could not find a \'AzureWebJobsStorage\' application setting')
+
+    container_name = "function-releases"
+    blob_name = "{}-{}.zip".format(datetime.datetime.today().strftime('%Y%m%d%H%M%S'), str(uuid.uuid4()))
+
+    block_blob_service = BlockBlobService(connection_string=storage_connection)
+    if not block_blob_service.exists(container_name):
+        block_blob_service.create_container(container_name)
+
+    # https://gist.github.com/vladignatyev/06860ec2040cb497f0f3
+    def progress_callback(current, total):
+        total_length = 30
+        filled_length = int(round(total_length * current) / float(total))
+        percents = round(100.0 * current / float(total), 1)
+        progress_bar = '=' * filled_length + '-' * (total_length - filled_length)
+        print('\r{} {}%'.format(progress_bar, percents), flush=True, end="")
+
+    print('Uploading...')
```

Comment:
Use Logger.warning instead

Reply:
I used print statements because we need to show the users information and these are not warnings. And it would be difficult to show a progress bar using logger.
Additionally, we don't need to show them any json.

Reply:
logger.warning should work the same as print -> you don't need a JSON object as an input  to use logger.warning. You can see webapp up command which uses this for general text display

Reply:
Discussed offline with @panchagnula, I have print there to show the progress bar. To do that, I need to be able to print directly to stdout as well as reset the cursor in the terminal. I am unable to do that using `logger.warning`.

@yugangw-msft, we were wondering what's your take on it? Do you think it's ok to use `print` here given it's not an ARM call, and need a progress bar.

Reply:
I agree with @panchagnula,  `print` should be avoided as the the principle here is anything going to `stdout` should be data which can be scripted. 
Any reason you won't want to use CLI's built-in progress support like the [code here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/extension/operations.py#L82)

Reply:
I wasn't aware of the built-in progress :)
I have updated the PR to use that. With that, I wouldn't need the `print` statements. I got rid of those. Thank you both. Glad it worked out